### DISCUSSION TITLE
Why refit cares too much about url path's starting charecter?

### DISCUSSION THREAD

**Discussion by yasmoradi** [reactiveui/refit#1604] [Q&A] [answered]
Exception's message that I've got:
URL path `api/Category/Get` must start with `'/'` and be of the form `'/foo/bar/baz'`

What's the benefit of this exception? How refit knows about my http client's base address? what if my base address is something like `https://api.mycompany.com/v1/` with `/` character at the end?

**Comment by glennawatson** [comment, marked as answer]
The leading `/` is just a syntax convention Refit enforces on the path you put in the attribute. It does not mean Refit ignores your base address.

What actually happens:
- Refit reads `HttpClient.BaseAddress.AbsolutePath` (for example `/v1/` from `https://api.mycompany.com/v1/`) and prepends it to your method path.
- So your base path is preserved. You do not lose `/v1/`.

For your case:
- Base address: `https://api.mycompany.com/v1/` (keep the trailing slash).
- Attribute: `[Get("/api/Category/Get")]` (with the leading slash, to satisfy the validator).
- Final URL: `https://api.mycompany.com/v1/api/Category/Get`.

So the exception is just telling you to write the attribute path in the canonical `/foo/bar/baz` form. It is not assuming anything about your host or stripping your base path. The benefit of the check is to catch accidental relative paths and to keep the combined URL deterministic across configurations.

Reference: VerifyUrlPathIsSane in RestMethodInfo.cs (the validator) and the path is combined with `BaseAddress.AbsolutePath` in RequestBuilderImplementation.cs.
